src/Tools/DTO/Tool.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Tools\DTO;
+
+use WordPress\AiClient\Common\Contracts\WithJsonSchema;
+use WordPress\AiClient\Providers\Enums\ToolTypeEnum;
+
+/**
+ * Represents a tool configuration for AI models.
+ *
+ * Tools allow AI models to perform actions beyond text generation,
+ * such as calling functions or performing web searches.
+ *
+ * @since n.e.x.t
+ */
+class Tool implements WithJsonSchema
+{
+    /**
+     * @var ToolTypeEnum The type of tool.
+     */
+    private ToolTypeEnum $type;
+
+    /**
+     * @var FunctionDeclaration[]|null Function declarations (when type is FUNCTION_DECLARATIONS).
+     */
+    private ?array $functionDeclarations = null;
+
+    /**
+     * @var WebSearch|null Web search configuration (when type is WEB_SEARCH).
+     */
+    private ?WebSearch $webSearch = null;
+
+    /**
+     * Private constructor to enforce factory method usage.
+     *
+     * @since n.e.x.t
+     *
+     * @param ToolTypeEnum $type The type of tool.
+     */
+    private function __construct(ToolTypeEnum $type)
+    {
+        $this->type = $type;
+    }
```

Comment:
See my feedback on `MessagePart`, it applies here similarly: We should allow passing in any of the supported data structures and infer the type from that.

Reply:
Resolved in a80601e